File: 23/1/1.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    string line, number;
    number.resize(2);
    int total = 0;
    int i;
    while(getline(cin, line)) {
        i = 0;
        while (!isdigit(line[i])) { i++; }
        number[0] = line[i];

        i = line.length() - 1;
        while (!isdigit(line[i])) { i--; }
        number[1] = line[i];

        total += stoi(number);
    }
    cout << total;
    return 0;
}

File: 23/3/1.py
from enum import Enum
import sys

class T(Enum):
    IGNORE = 1
    DIGIT = 2
    SYMBOL = 3

def get_type(char : str):
    if char == '.': return T.IGNORE
    if char.isdigit(): return T.DIGIT
    return T.SYMBOL

moves = [
    (-1, 1), (0, 1), (1, 1),
    (-1, 0),         (1, 0),
    (-1,-1), (0,-1), (1,-1)]

total = 0
number = ''
found_symbol = False

grid = list(map(str.rstrip, sys.stdin.readlines()))
slack = '.' * (len(grid[0]) + 2)
grid = [slack] + [ '.' + line + '.' for line in grid ] + [slack]

for i in range(1, len(grid)):
    for j in range(1, len(grid)):
        char = grid[i][j]
        if get_type(char) == T.DIGIT:
            number += char
            if not found_symbol:
                for x, y in moves:
                    if get_type(grid[i+y][j+x]) == T.SYMBOL:
                        found_symbol = True
                        break
        else:
            if number != '':
                if found_symbol: 
                    found_symbol = False
                    total += int(number)     
                number = ''

print(total)